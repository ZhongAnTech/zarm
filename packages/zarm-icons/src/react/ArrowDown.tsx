import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowDown = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M494.094 733.426a41.472 41.472 0 01-20.023-11.126L140.16 388.388c-16.272-16.271-16.272-42.653 0-58.925 16.271-16.272 42.653-16.272 58.925 0L503.66 634.037 808.186 329.51c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L533.2 722.347c-10.607 10.607-25.508 14.3-39.106 11.08z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowDown);
export default ForwardRef;
