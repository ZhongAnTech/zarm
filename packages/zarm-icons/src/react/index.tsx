export { default as ArrowDown } from './ArrowDown';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as ArrowUp } from './ArrowUp';
export { default as CloseCircleFill } from './CloseCircleFill';
export { default as CloseCircle } from './CloseCircle';
export { default as Close } from './Close';
export { default as DeleteKey } from './DeleteKey';
export { default as Empty } from './Empty';
export { default as Keyboard } from './Keyboard';
export { default as Minus } from './Minus';
export { default as PlusCircle } from './PlusCircle';
export { default as Plus } from './Plus';
export { default as Search } from './Search';
export { default as SuccessCircle } from './SuccessCircle';
export { default as Success } from './Success';
export { default as Volume } from './Volume';
export { default as WarningCircle } from './WarningCircle';
export { default as Warning } from './Warning';
