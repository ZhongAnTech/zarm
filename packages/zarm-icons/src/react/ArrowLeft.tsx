import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowLeft = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowLeft);
export default ForwardRef;
