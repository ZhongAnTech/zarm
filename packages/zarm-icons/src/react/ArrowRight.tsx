import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowRight = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowRight);
export default ForwardRef;
