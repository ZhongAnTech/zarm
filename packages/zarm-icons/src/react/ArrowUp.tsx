import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowUp = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M513.176 285.05a41.472 41.472 0 0120.024 11.127L867.11 630.089c16.272 16.271 16.272 42.653 0 58.925s-42.654 16.272-58.925 0L503.612 384.44 199.085 688.967c-16.272 16.271-42.654 16.271-58.925 0-16.272-16.272-16.272-42.654 0-58.926L474.07 296.129c10.606-10.606 25.508-14.299 39.105-11.078z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowUp);
export default ForwardRef;
