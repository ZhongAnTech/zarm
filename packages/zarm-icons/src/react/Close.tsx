import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgClose = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M560.149 501.223l255.344 255.344c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L501.223 560.149 245.88 815.493c-16.272 16.272-42.654 16.272-58.925 0-16.272-16.272-16.272-42.654 0-58.926l255.344-255.344L186.954 245.88c-16.272-16.272-16.272-42.654 0-58.925 16.271-16.272 42.653-16.272 58.925 0l255.344 255.344 255.344-255.344c16.272-16.272 42.654-16.272 58.926 0 16.272 16.271 16.272 42.653 0 58.925L560.149 501.223z"
          fillRule="nonzero"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgClose);
export default ForwardRef;
