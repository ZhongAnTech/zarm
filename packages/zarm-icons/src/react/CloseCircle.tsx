import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgCloseCircle = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <g fillRule="evenodd">
          <g>
            <path d="M500 460.716l137.493-137.493c10.848-10.848 28.436-10.848 39.284 0l.101.102c10.735 10.848 10.645 28.344-.203 39.08l-138.207 136.78 138.309 138.308c10.848 10.848 10.848 28.436 0 39.284-10.848 10.848-28.436 10.848-39.284 0L498.98 538.264 359.85 675.956c-10.41 10.303-27.188 10.26-37.545-.097-10.34-10.341-10.34-27.107 0-37.448L460.716 500 323.223 362.507c-10.848-10.848-10.848-28.436 0-39.284 10.848-10.848 28.436-10.848 39.284 0L500 460.716z" />
            <path
              d="M190.64 809.36c170.855 170.854 447.865 170.854 618.72 0 170.854-170.855 170.854-447.865 0-618.72-170.855-170.854-447.865-170.854-618.72 0-170.854 170.855-170.854 447.865 0 618.72zm-44.193 44.193c-195.263-195.262-195.263-511.844 0-707.106 195.262-195.263 511.844-195.263 707.106 0 195.263 195.262 195.263 511.844 0 707.106-195.262 195.263-511.844 195.263-707.106 0z"
              fillRule="nonzero"
            />
          </g>
        </g>
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgCloseCircle);
export default ForwardRef;
