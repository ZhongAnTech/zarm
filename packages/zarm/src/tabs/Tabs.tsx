import * as React from 'react';
import classnames from 'classnames';
import TabPanel from './TabPanel';
import Carousel from '../carousel';
import type { CarouselHTMLElement } from '../carousel';
import { getTransformPropValue, getPxStyle } from './util/index';
import { scrollTo } from '../utils/dom';
import { ConfigContext } from '../n-config-provider';
import type { TabPanelProps } from './TabPanel';
import type { BaseTabsProps } from './interface';

const getChildChecked = (children: TabPanelProps['children']) => {
  let selectIndex;
  React.Children.forEach(children, (item, index) => {
    if (React.isValidElement(item) && item.props && item.props.selected) {
      selectIndex = index;
    }
  });
  return selectIndex;
};

const parseValueBoundary = (
  inputValue: TabsProps['value'],
  children: TabPanelProps['children'],
) => {
  const count = React.Children.count(children);
  if (inputValue! <= 0) {
    return 0;
  }
  if (inputValue! > count - 1) {
    return count - 1;
  }
  return inputValue;
};

const getValue = (props: TabsProps, defaultValue: TabsProps['value']) => {
  if (typeof props.value !== 'undefined') {
    return parseValueBoundary(props.value, props.children);
  }
  if (typeof props.defaultValue !== 'undefined') {
    return parseValueBoundary(props.defaultValue, props.children);
  }
  if (getChildChecked(props.children)) {
    return parseValueBoundary(getChildChecked(props.children), props.children);
  }
  return parseValueBoundary(defaultValue, props.children);
};

export interface TabsProps extends BaseTabsProps {
  className?: string;
}

interface CompoundedComponent
  extends React.ForwardRefExoticComponent<TabsProps & React.RefAttributes<HTMLDivElement>> {
  Panel: typeof TabPanel;
}

const Tabs = React.forwardRef<unknown, TabsProps>((props, ref) => {
  const {
    className,
    value,
    defaultValue,
    disabled,
    swipeable,
    scrollable,
    direction,
    lineWidth,
    onChange,
    children,
  } = props;

  const tabsRef = (ref as any) || React.createRef<HTMLDivElement>();
  const carouselRef = React.useRef<CarouselHTMLElement>(null);
  const tablistRef = React.useRef<HTMLUListElement>(null);
  const [itemWidth, setItemWidth] = React.useState(0);
  const [currentValue, setCurrentValue] = React.useState(
    getValue({ value, defaultValue, children }, 0),
  );

  const { prefixCls: globalPrefixCls } = React.useContext(ConfigContext);
  const prefixCls = `${globalPrefixCls}-tabs`;

  const isVertical: boolean = direction === 'vertical';

  const parseValue = React.useCallback((inputValue) => parseValueBoundary(inputValue, children), [
    children,
  ]);

  const cls = classnames(prefixCls, className, `${prefixCls}--${direction}`, {
    [`${prefixCls}--scroll`]: scrollable,
  });

  // 计算 line 大小和位置
  const caclLineSizePos = () => {
    const newValue = parseValue(currentValue)!;
    const ChildCount = React.Children.count(children);

    let pos = 100 * newValue;
    if (scrollable && tablistRef.current) {
      const el = tablistRef.current.children[newValue];
      const { offsetLeft = 0, offsetTop = 0 } = el as HTMLElement;
      pos = isVertical ? offsetTop : offsetLeft;
    }

    const size = scrollable ? `${itemWidth}px` : `${100 / ChildCount}%`;
    const transformValue = scrollable
      ? getPxStyle(pos, 'px', isVertical)
      : getPxStyle(pos, '%', isVertical);
    const styleUl = getTransformPropValue(transformValue);
    const itemSize = isVertical ? { height: `${size}` } : { width: `${size}` };

    return {
      ...styleUl,
      ...itemSize,
    };
  };

  const onTabChange = React.useCallback(
    (newValue: number) => {
      if (typeof value === 'undefined') {
        setCurrentValue(newValue);
      }
      typeof onChange === 'function' && onChange(newValue);
    },
    [value, onChange],
  );

  const onTabClick = (tab: React.ReactElement<TabPanelProps, typeof TabPanel>, index: number) => {
    if (disabled || tab.props.disabled) {
      return;
    }
    if (swipeable) {
      carouselRef.current && carouselRef.current!.onSlideTo(index);
      return;
    }
    onTabChange(index);
  };

  const lineStyle: React.CSSProperties = caclLineSizePos();
  let lineInnerRender;
  if (lineWidth) {
    lineStyle.backgroundColor = 'transparent';
    lineInnerRender = <span className={`${prefixCls}__line__inner`} style={{ width: lineWidth }} />;
  }

  // 渲染内容
  let contentRender;
  if (swipeable) {
    contentRender = (
      <Carousel
        swipeable={!disabled}
        direction={direction === 'vertical' ? 'up' : 'left'}
        showPagination={false}
        activeIndex={parseValue(currentValue)}
        ref={carouselRef}
        onChange={onTabChange}
      >
        {React.Children.map(children, (item: any, index: number) => (
          <div key={+index}>{item.props.children}</div>
        ))}
      </Carousel>
    );
  } else {
    contentRender = React.Children.map(
      children,
      (item: React.ReactElement<TabPanelProps, typeof TabPanel>, index: number) => (
        <TabPanel {...item.props} selected={parseValue(currentValue) === index} />
      ),
    );
  }

  const renderTabs = (tab: React.ReactElement<TabPanelProps, typeof TabPanel>, index: number) => {
    const itemCls = classnames(`${prefixCls}__tab`, tab.props.className, {
      [`${prefixCls}__tab--disabled`]: disabled || tab.props.disabled,
      [`${prefixCls}__tab--active`]: parseValue(currentValue) === index,
    });

    return (
      <li role="tab" key={+index} className={itemCls} onClick={() => onTabClick(tab, index)}>
        {tab.props.title}
      </li>
    );
  };

  // 渲染选项
  const tabsRender = React.Children.map(children, renderTabs);

  const getItemStyle = (el, prop) => {
    let newValue = '0';
    if (prop in el.style) {
      newValue = el.style[prop] || window.getComputedStyle(el).getPropertyValue(prop) || '0';
    }
    return newValue;
  };

  const calculateLineWidth = React.useCallback(() => {
    if (!scrollable) {
      return;
    }
    const newValue = parseValue(currentValue)!;
    const el = tablistRef.current!.children[newValue];
    const size = isVertical ? getItemStyle(el, 'height') : getItemStyle(el, 'width');

    setItemWidth(parseInt(size.toString(), 10));
  }, [parseValue, currentValue, isVertical, scrollable]);

  const calculateScorllLeftLocation = React.useCallback(() => {
    if (!scrollable) {
      return false;
    }
    const newValue = parseValue(currentValue)!;
    const index = newValue - 1 >= 0 ? newValue - 1 : 0;
    const prevTabItem = tablistRef.current!.childNodes[index];
    if (scrollable && tablistRef.current && prevTabItem) {
      const { offsetTop: top = 0, offsetLeft: left = 0 } = prevTabItem as HTMLElement;
      scrollTo(tablistRef.current, top, left, 0.3);
    }
  }, [parseValue, currentValue, scrollable]);

  React.useEffect(() => {
    if (React.Children.count(children)) {
      calculateLineWidth();
      calculateScorllLeftLocation();
    }
  }, [calculateLineWidth, calculateScorllLeftLocation, children]);

  React.useEffect(() => {
    setCurrentValue(getValue({ value, defaultValue, children }, 0));
  }, [value, defaultValue, children]);

  return (
    <div className={cls} ref={tabsRef}>
      <div className={`${prefixCls}__header`}>
        <ul className={`${prefixCls}__tablist`} role="tablist" ref={tablistRef}>
          {tabsRender}
          <div className={`${prefixCls}__line`} style={lineStyle}>
            {lineInnerRender}
          </div>
        </ul>
      </div>
      <div className={`${prefixCls}__body`}>{contentRender}</div>
    </div>
  );
}) as CompoundedComponent;

Tabs.displayName = 'Tabs';

Tabs.defaultProps = {
  disabled: false,
  swipeable: false,
  scrollable: false,
  direction: 'horizontal',
};

export default Tabs;
