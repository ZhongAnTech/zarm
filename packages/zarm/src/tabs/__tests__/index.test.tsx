import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Tabs from '../index';

describe('Tab', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Tabs>
        <Tabs.Panel title="选项卡1">
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly vertical', () => {
    const { container } = render(
      <Tabs direction="vertical">
        <Tabs.Panel title="选项卡1">
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    expect(container).toMatchSnapshot();
  });

  it('canSwipe', () => {
    const { container } = render(
      <Tabs swipeable>
        <Tabs.Panel title="选项卡1">
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    expect(container).toMatchSnapshot();
  });

  it('lineWidth is auto', () => {
    const { container } = render(
      <Tabs lineWidth="auto">
        <Tabs.Panel title="选项卡1">
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    expect(container).toMatchSnapshot();
  });

  it('receive new value', () => {
    const onChange = jest.fn();
    const { container } = render(
      <Tabs onChange={onChange} value={1}>
        <Tabs.Panel title="选项卡1">
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    expect(container).toMatchSnapshot();
  });

  it('click tabs', () => {
    const onChange = jest.fn();
    const { container } = render(
      <Tabs scrollable onChange={onChange}>
        <Tabs.Panel title="选项卡1" disabled>
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡3">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    const el = container.querySelectorAll('.za-tabs__tab');
    fireEvent.click(el[1]);
    expect(onChange).toBeCalledWith(1);
    const last = el[el.length - 1];
    fireEvent.click(last);
    expect(onChange).toBeCalledWith(2);
  });

  it('scroll tabs', () => {
    const onChange = jest.fn();
    const { container } = render(
      <Tabs swipeable onChange={onChange} scrollable>
        <Tabs.Panel title="选项卡1" disabled>
          <div>试试点我左滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡2">
          <div>试试点我右滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡3">
          <div>试试点我右滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡4">
          <div>试试点我右滑</div>
        </Tabs.Panel>
        <Tabs.Panel title="选项卡5">
          <div>试试点我右滑</div>
        </Tabs.Panel>
      </Tabs>,
    );
    const el = container.querySelectorAll('.za-tabs__tab');
    const last = el[el.length - 1];
    fireEvent.click(last);
    expect(onChange).toBeCalledWith(4);
  });
});
