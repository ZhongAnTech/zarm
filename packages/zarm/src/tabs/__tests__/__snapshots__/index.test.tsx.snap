// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab canSwipe 1`] = `
<div>
  <div
    class="za-tabs za-tabs--horizontal"
  >
    <div
      class="za-tabs__header"
    >
      <ul
        class="za-tabs__tablist"
        role="tablist"
      >
        <li
          class="za-tabs__tab za-tabs__tab--active"
          role="tab"
        >
          选项卡1
        </li>
        <li
          class="za-tabs__tab"
          role="tab"
        >
          选项卡2
        </li>
        <div
          class="za-tabs__line"
          style="transform: translate3d(0%, 0px, 0px); width: 50%;"
        />
      </ul>
    </div>
    <div
      class="za-tabs__body"
    >
      <div
        class="za-carousel za-carousel--horizontal"
      >
        <div
          class="za-carousel__items"
          style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
        >
          <div
            class="za-carousel__item"
          >
            <div>
              试试点我左滑
            </div>
          </div>
          <div
            class="za-carousel__item"
          >
            <div>
              试试点我右滑
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tab lineWidth is auto 1`] = `
<div>
  <div
    class="za-tabs za-tabs--horizontal"
  >
    <div
      class="za-tabs__header"
    >
      <ul
        class="za-tabs__tablist"
        role="tablist"
      >
        <li
          class="za-tabs__tab za-tabs__tab--active"
          role="tab"
        >
          选项卡1
        </li>
        <li
          class="za-tabs__tab"
          role="tab"
        >
          选项卡2
        </li>
        <div
          class="za-tabs__line"
          style="transform: translate3d(0%, 0px, 0px); width: 50%; background-color: transparent;"
        >
          <span
            class="za-tabs__line__inner"
            style="width: auto;"
          />
        </div>
      </ul>
    </div>
    <div
      class="za-tabs__body"
    >
      <div
        class="za-tabs__panel za-tabs__panel--active"
        role="tabpanel"
      >
        <div>
          试试点我左滑
        </div>
      </div>
      <div
        class="za-tabs__panel"
        role="tabpanel"
      >
        <div>
          试试点我右滑
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tab receive new value 1`] = `
<div>
  <div
    class="za-tabs za-tabs--horizontal"
  >
    <div
      class="za-tabs__header"
    >
      <ul
        class="za-tabs__tablist"
        role="tablist"
      >
        <li
          class="za-tabs__tab"
          role="tab"
        >
          选项卡1
        </li>
        <li
          class="za-tabs__tab za-tabs__tab--active"
          role="tab"
        >
          选项卡2
        </li>
        <div
          class="za-tabs__line"
          style="transform: translate3d(100%, 0px, 0px); width: 50%;"
        />
      </ul>
    </div>
    <div
      class="za-tabs__body"
    >
      <div
        class="za-tabs__panel"
        role="tabpanel"
      >
        <div>
          试试点我左滑
        </div>
      </div>
      <div
        class="za-tabs__panel za-tabs__panel--active"
        role="tabpanel"
      >
        <div>
          试试点我右滑
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tab renders correctly 1`] = `
<div>
  <div
    class="za-tabs za-tabs--horizontal"
  >
    <div
      class="za-tabs__header"
    >
      <ul
        class="za-tabs__tablist"
        role="tablist"
      >
        <li
          class="za-tabs__tab za-tabs__tab--active"
          role="tab"
        >
          选项卡1
        </li>
        <li
          class="za-tabs__tab"
          role="tab"
        >
          选项卡2
        </li>
        <div
          class="za-tabs__line"
          style="transform: translate3d(0%, 0px, 0px); width: 50%;"
        />
      </ul>
    </div>
    <div
      class="za-tabs__body"
    >
      <div
        class="za-tabs__panel za-tabs__panel--active"
        role="tabpanel"
      >
        <div>
          试试点我左滑
        </div>
      </div>
      <div
        class="za-tabs__panel"
        role="tabpanel"
      >
        <div>
          试试点我右滑
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tab renders correctly vertical 1`] = `
<div>
  <div
    class="za-tabs za-tabs--vertical"
  >
    <div
      class="za-tabs__header"
    >
      <ul
        class="za-tabs__tablist"
        role="tablist"
      >
        <li
          class="za-tabs__tab za-tabs__tab--active"
          role="tab"
        >
          选项卡1
        </li>
        <li
          class="za-tabs__tab"
          role="tab"
        >
          选项卡2
        </li>
        <div
          class="za-tabs__line"
          style="transform: translate3d(0px, 0%, 0px); height: 50%;"
        />
      </ul>
    </div>
    <div
      class="za-tabs__body"
    >
      <div
        class="za-tabs__panel za-tabs__panel--active"
        role="tabpanel"
      >
        <div>
          试试点我左滑
        </div>
      </div>
      <div
        class="za-tabs__panel"
        role="tabpanel"
      >
        <div>
          试试点我右滑
        </div>
      </div>
    </div>
  </div>
</div>
`;
